modules/product/src/main/java/com/opengamma/strata/product/SecurityId.java

```diff
@@ -0,0 +1,205 @@
+   * <p>
+   * This parses the identifier from the form produced by {@code toString()}
+   * which is '{@code $scheme~$value}'.
+   *
+   * @param str  the identifier to parse
+   * @return the security identifier
+   * @throws IllegalArgumentException if the identifier cannot be parsed
+   */
+  @FromString
+  public static SecurityId parse(String str) {
+    return new SecurityId(StandardId.parse(str));
+  }
+
+  // creates an identifier for a combined calendar
+  private SecurityId(StandardId standardId) {
+    this.standardId = ArgChecker.notNull(standardId, "standardId");
+  }
+
+  // resolve after deserialization
+  private Object readResolve() {
+    return of(standardId);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the standard two-part identifier.
+   * 
+   * @return the standard two-part identifier
+   */
+  public StandardId getStandardId() {
+    return standardId;
+  }
+
+  /**
+   * Gets the type of data this identifier refers to.
+   * <p>
+   * A {@code SecurityId} refers to a {@code ReferenceSecurity}.
+   *
+   * @return the type of the reference data this identifier refers to
+   */
+  @Override
+  public Class<ReferenceSecurity> getReferenceDataType() {
+    return ReferenceSecurity.class;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Resolves this identifier to a security using the specified reference data.
+   * <p>
+   * This returns an instance of {@link ReferenceSecurity}.
+   * <p>
+   * Resolved objects may be bound to data that changes over time.
+   * If the data changes, the resolved form will not be updated.
+   * Care must be taken when placing the resolved form in a cache or persistence layer.
+   * 
+   * @param refData  the reference data, used to resolve the reference
+   * @return the resolved security
+   * @throws ReferenceDataNotFoundException if the identifier is not found
+   */
+  @Override
+  public ReferenceSecurity resolve(ReferenceData refData) {
+    return refData.getValue(this);
+  }
+
+  /**
+   * Resolves this identifier to a security using the specified reference data and type.
+   * <p>
+   * This returns an instance of {@link ReferenceSecurity} cast to the specified type.
+   * <p>
+   * Resolved objects may be bound to data that changes over time.
+   * If the data changes, the resolved form will not be updated.
+   * Care must be taken when placing the resolved form in a cache or persistence layer.
+   * 
+   * @param <T>  the type of the security
+   * @param refData  the reference data, used to resolve the reference
+   * @param securityType  the type of the security
+   * @return the resolved security
+   * @throws ReferenceDataNotFoundException if the identifier is not found
+   * @throws ClassCastException if the identifier exists, but is not of the correct type
+   */
+  public <T extends ReferenceSecurity> T resolve(ReferenceData refData, Class<T> securityType) {
```

Comment:
Does this method add enough to justify its place on the API? Effectively it enables this:

    MySecurity security = securityId.resolve(refData, MySecurity.class);

instead of:

    MySecurity security = (MySecurity) securityId.resolve(refData);

Is it worth cluttering up the API for that?